src/Microsoft.SymbolStore.Client/FileSemaphore.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.SymbolStore.Client
+{
+    public static class FileSemaphore
+    {
+        class LockedFile : IDisposable
+        {
+            private bool _disposed = false;
+            private string _lockPath;
+            private FileStream _fileStream;
+
+            public LockedFile(string lockPath, FileStream fs)
+            {
+                _lockPath = lockPath;
+                _fileStream = fs;
+            }
+
+            #region IDisposable Support
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!_disposed)
+                {
+                    if (disposing)
+                    {
+                        _fileStream.Dispose();
+                    }
+
+                    try
+                    {
+                        File.Delete(_lockPath);
+                    }
+                    catch (IOException)
+                    {
+                    }
+                    
+                    _disposed = true;
+                }
+            }
+            
+            ~LockedFile()
+            {
+               Dispose(false);
```

Comment:
Rather than disposing the file I'd suggest asserting. I can't imagine its ever correct to forget to call Dispose on this type explicitly. (Doing stuff in finalizers is also a pain, for example I think its legal for _lockPath to already be GC'ed at this point)